tgui/packages/tgui/interfaces/PersonalCrafting.tsx

```diff
@@ -0,0 +1,318 @@
+  && (!data.dm_subcategory || data.dm_subcategory === item.dm_subcategory)
+);
+
+export const PersonalCrafting = (props, context) => {
+  const { act, data: rawData } = useBackend<RawData>(context);
+  const data = remapData(rawData);
+
+  const {
+    busy,
+    display_craftable_only,
+    display_compact,
+    recipes,
+    groups,
+  } = data;
+
+  const shownRecipes = flow([
+    filter<Recipe>((recipe) => (
+      // Show selected category only
+      isCategorySelected(data, recipe)
+      // If craftable only is selected, then filter by craftability
+      && (!display_craftable_only || recipe.craftable)
+    )),
+    sortBy<Recipe>((recipe) => [
+      -recipe.craftable,
+      recipe.name,
+    ]),
+  ])(recipes);
+
+  return (
+    <Window title="Crafting Menu" width={700} height={700}>
+      <Window.Content>
+        <Stack fill>
+          <Stack.Item width="150px">
+            <Section fill scrollable title="Category">
+              {groups.map((group) => (
+                <Section key={group.name} title={group.name}>
+                  {group.categories.map((category) => (
+                    <Button
+                      key={category.name}
+                      fluid
+                      color="transparent"
+                      selected={isCategorySelected(data, category)}
+                      onClick={() => act('set_category', {
+                        category: category.dm_category,
+                        subcategory: category.dm_subcategory,
+                      })}>
+                      {category.name}
+                    </Button>
+                  ))}
+                </Section>
+              ))}
+            </Section>
+          </Stack.Item>
+          <Stack.Item grow={3}>
+            <Section
+              fill
+              title="Recipes"
+              buttons={
+                <>
+                  <Button.Checkbox
+                    content="Compact"
+                    checked={display_compact}
+                    onClick={() => act('toggle_compact')}
+                  />
+                  <Button.Checkbox
+                    content="Craftable Only"
+                    checked={display_craftable_only}
+                    onClick={() => act('toggle_recipes')}
+                  />
+                </>
+              }>
+              <Section fill scrollable>
+                {busy ? (
+                  <Dimmer fontSize="32px">
+                    <Icon name="cog" spin={1} />
+                    {' Crafting...'}
+                  </Dimmer>
+                ) : (
+                  <CraftingList
+                    recipes={shownRecipes}
+                    compact={Boolean(display_compact)} />
+                )}
+              </Section>
+            </Section>
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+type CraftingListProps = {
+  recipes: Recipe[];
+  // eslint-disable-next-line react/no-unused-prop-types
```

Comment:
What's this for? It looks like you use it.

Reply:
This ESLint rule from react preset is bugged.

Reply:
I should probably disable it globally for this reason.